### PR TITLE
feat(datastore): Support overriding of service endpoint

### LINKED ISSUE #3753: [datastore] Support overriding endpoint host
Support overriding the endpoint host in the veneer layer, by setting either constructor argument or config field.

See #3752 for an example from bigtable.

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #3753]

### CODE REVIEW

File: google-cloud-datastore/lib/google/cloud/datastore.rb [hunk 1]
```diff
@@ -98,12 +100,13 @@ module Datastore
...
       def self.new project_id: nil, credentials: nil, scope: nil, timeout: nil,
-                   client_config: nil, emulator_host: nil, project: nil,
-                   keyfile: nil
+                   client_config: nil, endpoint: nil, emulator_host: nil,
+                   project: nil, keyfile: nil
         project_id    ||= (project || default_project_id)
         scope         ||= configure.scope
         timeout       ||= configure.timeout
         client_config ||= configure.client_config
+        endpoint ||= configure.endpoint
```
Comment (dazuma) [inline google-cloud-datastore/lib/google/cloud/datastore.rb]: Can you either make the whitespace match the formatting around it, or remove the fancy formatting in the other lines?

File: google-cloud-datastore/test/google/cloud/datastore_test.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@
   describe "#datastore" do
     it "calls out to Google::Cloud.datastore" do
       gcloud = Google::Cloud.new
-      stubbed_datastore = ->(project, keyfile, scope: nil, timeout: nil, client_config: nil) {
+      stubbed_datastore = ->(project, keyfile, scope: nil, timeout: nil, host: nil, client_config: nil) {
```
Comment (dazuma) [inline google-cloud-datastore/test/google/cloud/datastore_test.rb]: Check that host is nil (same as the other fields)

Comment (quartzmo) [reply google-cloud-datastore/test/google/cloud/datastore_test.rb]: Done!

File: google-cloud-datastore/lib/google/cloud/datastore.rb (hunk 1 above)
Comment (quartzmo) [reply google-cloud-datastore/lib/google/cloud/datastore.rb]: Done!

**Pull request merged**